src/Integration/YoastWooCommerceSeo.php

```diff
@@ -76,6 +76,19 @@ function ( $sources, $attribute_id ) {
 			10,
 			2
 		);
+
+		add_filter(
+			'woocommerce_gla_gtin_migration_value',
+			function ( $gtin, $product ) {
+				if ( ! $gtin ) {
+					return $this->get_gtin( self::VALUE_KEY, $product );
```

Comment:
This part is not working for me. Because when a product was previously setup the `_wc_gla_gtin` had the value of `yoast_seo` and then the `get_gtin` function would replace it with the value stored by Yoast.

However here we only enter the function if no GTIN number was found. So we'll need to decide, do we want to only migrate if the GLA value is set to `yoast_seo` or do we want to migrate whenever we have a value stored by Yoast.

If it's the latter then we should change it to something like:
```suggestion
				if ( ! $gtin || self::VALUE_KEY === $gtin ) {
					return $this->get_gtin( self::VALUE_KEY, $product );
```

Otherwise we could simplify the whole function as long as we pass a string to `get_gtin` it will check the prefix:
```php
		add_filter(
			'woocommerce_gla_gtin_migration_value',
			function ( $gtin, $product ) {
				return $this->get_gtin( (string) $gtin, $product );
			},
			10,
			2
		);
```

Reply:
> if ( ! $gtin || self::VALUE_KEY === $gtin ) {
					return $this->get_gtin( self::VALUE_KEY, $product );

This will always fail because $gtin is actually the `$gtin` number. `self::VALUE_KEY` is a key reference and is only necessary because 
`YoastWooCommerceSeo::get_gtin` [requires that ](https://github.com/woocommerce/google-listings-and-ads/pull/2677/files#diff-c1a51c2c069efcf71305ae9085386da9a893cc2b9ad78c70a22b42d3f0608c2dL112) because is doing a key check as it's the function is being used also in other contexts so we don't need to compare that with $gtin for our case. I believe 

What I'm doing with:

```php
add_filter(
			'woocommerce_gla_gtin_migration_value',
			function ( $gtin, $product ) {
				if ( ! $gtin ) {
					return $this->get_gtin( self::VALUE_KEY, $product );
				}

				return $gtin;
			},
			10,
			2
		);
```

It is to be sure the value is not being overriden. In other words. If the GTIN is set in Google for WooCommerce we won't override it with YOAST GTIN. We only get YOAST value if GTIN is null or empty. Does that make sense?